lib/spark/dsl/extension.ex

```diff
@@ -1484,7 +1484,10 @@ defmodule Spark.Dsl.Extension do
                   Spark.CodeHelpers.lift_functions(arg_value, key, __CALLER__)
 
                 if is_nil(arg_value) &&
-                     Enum.any?(entity.args, &(elem(&1, 0) == :optional && elem(&1, 1) == key)) do
+                     Enum.any?(
+                       entity.args,
+                       &(is_tuple(&1) && elem(&1, 0) == :optional && elem(&1, 1) == key)
```

Comment:
Should this be `&1 == key || (&(is_tuple(&1) && elem(&1, 0) == :optional && elem(&1, 1) == key))`? Is this only meant to see if there are optional arguments?

Reply:
nvm, its clear to me now :)

Reply:
It's only meant to catch optional arguments named `key`.

Reply:
It's a bit shit, but I don't want to add yet another function to this module.